libs/ledger-live-common/src/families/evm/editTransaction/getMinFees.ts

```diff
@@ -0,0 +1,68 @@
+import type { Transaction } from "@ledgerhq/coin-evm/types/index";
+import { getEnv } from "@ledgerhq/live-env";
+import { BigNumber } from "bignumber.js";
+
+/**
+ * Returns the minimum fees values to update a legacy transaction
+ * ⚠️ Makes sure the updated fees are at least 10% higher than the original fees
+ * since this is the minimum increase required by most nodes
+ */
+export const getMinLegacyFees = ({
+  gasPrice,
+}: {
+  gasPrice: BigNumber;
+}): { gasPrice: BigNumber } => {
+  const gasPriceFactor: number = getEnv("EVM_REPLACE_TX_LEGACY_GASPRICE_FACTOR");
+  return {
+    gasPrice: gasPrice.times(gasPriceFactor).integerValue(BigNumber.ROUND_CEIL),
+  };
+};
+
+/**
+ * Returns the minimum fees values to update an EIP1559 transaction
+ * ⚠️ Makes sure the updated fees are at least 10% higher than the original fees
+ * since this is the minimum increase required by most nodes
+ */
+export const getMinEip1559Fees = ({
+  maxFeePerGas,
+  maxPriorityFeePerGas,
+}: {
+  maxFeePerGas: BigNumber;
+  maxPriorityFeePerGas: BigNumber;
+}): {
+  maxFeePerGas: BigNumber;
+  maxPriorityFeePerGas: BigNumber;
+} => {
+  const maxFeePerGasFactor: number = getEnv("EVM_REPLACE_TX_EIP1559_MAXFEE_FACTOR");
+  const maxPriorityFeePerGasFactor: number = getEnv("EVM_REPLACE_TX_EIP1559_MAXPRIORITYFEE_FACTOR");
+
+  return {
+    maxFeePerGas: maxFeePerGas.times(maxFeePerGasFactor).integerValue(BigNumber.ROUND_CEIL),
+    maxPriorityFeePerGas: maxPriorityFeePerGas
+      .times(maxPriorityFeePerGasFactor)
+      .integerValue(BigNumber.ROUND_CEIL),
+  };
+};
+
+/**
+ * Returns the minimum fees values to update a evm transaction
+ * ⚠️ Makes sure the updated fees are at least 10% higher than the original fees
+ * since this is the minimum increase required by most nodes
+ */
+export const getMinFees = ({
+  transaction,
+}: {
+  transaction: Transaction;
+}): {
+  gasPrice?: BigNumber;
+  maxFeePerGas?: BigNumber;
+  maxPriorityFeePerGas?: BigNumber;
+} => {
+  if (transaction.type === 2) {
```

Comment:
Yeah I'm personally a bit confused what means type 1,2,3,...
I feel like an enum would make it simpler to know what we're talking about

Reply:
We already discussed this, but unfortunatley that's some necessary knowledge in order to contribute to the evm family. You can find some documentation [here](https://ethereum.org/fr/developers/docs/transactions/#typed-transaction-envelope) if needed listing the different types.
We can add an enum that will say that type 0 = legacy, type 1 = EIP-2718, type 2 = EIP-1559, but to me, it's not going to help anyone, as, if you don't know what type 1 or 2 are, then you don't know what EIP-2718 and EIP-1559 are.

Reply:
I'm for the enum as it provides keywords for further research

Reply:
Indeed I wasn't aware that `type 1` and `EIP-2930` could be used interchangeably and that it's a known term 👍🏿 Good for me then